src/FlaUILibrary/flaui/module/treeitems.py

```diff
@@ -0,0 +1,134 @@
+from FlaUILibrary.flaui.exception import FlaUiError
+
+class TreeItems:
+
+    def __init__(self,control):
+        self.Control = control
+        self.TreeItems = control.Items
+        self.CurrentTreeItem = None
+        self.SelectedTreeItem = self.Control.SelectedTreeItem
+        self.TreeItemNames = []
+        self.TreeItemsCount = 0
+        self.Location = []
+
+    def no_operation(self):
+        """ no_operation does nothing.
+        can be used to skip the line of the a function if it will be given as an argument.
+        """
+        pass
+    
+    def add_treeItems_count(self):
+        """ adds 1 to TreeItemsCount attribute of the created object.
+        """
+        self.TreeItemsCount +=1
+    
+    def select_treeItem_by_location(self, location):
+        """ Selects treeItem by location.
+
+        Args:
+            location (array): array of strings pointing the end item to be selected.
+        """
+        self.execute_by_location(location, "Select")
+    
+    def expand_treeItem_by_location(self, location):
+        """ Expands treeItem by location.
+
+        Args:
+            location (array): array of strings pointing the end item to be expanded.
+        """
+        self.execute_by_location(location, "Expand")
+
+    def collapse_treeItem_by_location(self, location):
+        """ Collapses treeItem by location.
+
+        Args:
+            location (array): array of strings pointing the end item to be collapsed.
+        """
+        self.execute_by_location(location, "Collapse")
+
+    def iterate_all_visible(self, funcMain, funcAfter):
+        for item in self.TreeItems:
+            state = item.ExpandCollapseState
+            self.CurrentTreeItem = item
+            funcMain()
+            #The element is LeafNode does not expand or collapse.
+            if state == 3:
+                continue
+            #The element is Expanded All children are visible.
+            elif state == 1:
+                self.TreeItems = item.Items
+                self.iterate_all_visible(funcMain, funcAfter)
+            #The element is PartiallyExpanded Some, but not all, children are visible..
+            elif state == 2:
+                self.TreeItems = item.Items
+                self.iterate_all_visible(funcMain, funcAfter)
+            #The element is Collapsed No children are visible.
+            elif state == 0:
+                continue
+            funcAfter()
```

Comment:
Remove the lambdas create for each type a seperate method and split up the method in a static context